chain-evm/src/machine.rs

```diff
@@ -568,19 +592,27 @@ impl<'a, State: EvmState> StackState<'a> for VirtualMachine<'a, State> {
     }
 
     fn set_code(&mut self, address: H160, code: Vec<u8>) {
-        self.substate.account_mut(address, self.state).code = code;
+        self.substate.account_mut(address, self.state).state.code = code;
     }
 
     fn transfer(&mut self, transfer: Transfer) -> Result<(), ExitError> {
         let source = self.substate.account_mut(transfer.source, self.state);
 
-        source.balance = match source.balance.checked_sub(transfer.value) {
+        source.balance = match source
+            .balance
+            .checked_sub(transfer.value.try_into().map_err(|_| ExitError::OutOfGas)?)
+        {
             Some(res) => res,
             None => return Err(ExitError::OutOfGas),
         };
 
         let target = self.substate.account_mut(transfer.target, self.state);
-        target.balance = match target.balance.checked_add(transfer.value) {
+        target.balance = match target.balance.checked_add(
+            transfer
+                .value
+                .try_into()
+                .map_err(|_| ExitError::Other("Balance overflow".into()))?,
```

Comment:
Please be consistent with error codes if the cause is the same, we can just bubble up the conversion error from `Balance` in this case